src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Cancellation.cs

```diff
@@ -79,20 +79,26 @@ public async Task PostAsync_CancelDuringRequestContentSend_TaskCanceledQuickly(b
         [MemberData(nameof(TwoBoolsAndCancellationMode))]
         public async Task GetAsync_CancelDuringResponseHeadersReceived_TaskCanceledQuickly(bool chunkedTransfer, bool connectionClose, CancellationMode mode)
         {
+            if (LoopbackServerFactory.IsHttp2 && (chunkedTransfer || connectionClose))
+            {
+                // There is no chunked encoding in HTTP/2
+                return;
+            }
+
             using (HttpClient client = CreateHttpClient())
             {
                 client.Timeout = Timeout.InfiniteTimeSpan;
                 var cts = new CancellationTokenSource();
 
-                await LoopbackServer.CreateServerAsync(async (server, url) =>
+                await LoopbackServerFactory.CreateServerAsync(async (server, url) =>
                 {
                     var partialResponseHeadersSent = new TaskCompletionSource<bool>();
                     var clientFinished = new TaskCompletionSource<bool>();
 
                     Task serverTask = server.AcceptConnectionAsync(async connection =>
                     {
-                        await connection.ReadRequestHeaderAndSendCustomResponseAsync(
-                            $"HTTP/1.1 200 OK\r\nDate: {DateTimeOffset.UtcNow:R}\r\n"); // missing final \r\n so headers don't complete
+                        await connection.ReadRequestDataAsync();
+                        await connection.SendResponseAsync(HttpStatusCode.OK,isFinal : false);
```

Comment:
Nit: previous test sent a Date header. Do we think that is immaterial?

Reply:
There is difference between many tests where HTTP/1 does send Date but HTTP/2 does not. If you look at GetAsync_EmptyResponseHeader_Success() change, we special case it for each protocol version. 
I don't think it matters in this test but we should decide and either do not send it unless needed or send it for both protocol versions. 
Rcommendations?

Reply:
It's weird that we special case this for GetAsync_EmptyResponseHeader_Success. 

We should be consistent in the implementation of our loopback servers. Either both add Date:, or neither do. I suspect we originally didn't add Date: for HTTP2 because we didn't have very sophisticated header encoding logic. But now we do, so we should be consistent. I don't have a strong opinion on which is best, though.

Reply:
ok. In that case I'm leaning towards adding it for both cases to make responses somewhat more realistic. It seems like real servers do use it in both cases.

Reply:
Seems reasonable to me